mgmt/smp/transport/smp_uart/src/smp_uart.c

```diff
@@ -354,42 +360,42 @@ nmgr_uart_rx_char(void *arg, uint8_t data)
         }
     }
     /* failed */
-    nus->nus_rx->omp_len = 0;
+    sus->sus_rx->omp_len = 0;
     m->om_len = 0;
     os_mbuf_free_chain(SLIST_NEXT(m, om_next));
     SLIST_NEXT(m, om_next) = NULL;
     return 0;
 }
 
 void
-nmgr_uart_pkg_init(void)
+smp_uart_pkg_init(void)
 {
-    struct nmgr_uart_state *nus = &nmgr_uart_state;
+    struct smp_uart_state *sus = &smp_uart_state;
     int rc;
     struct uart_conf uc = {
         .uc_speed = MYNEWT_VAL(NMGR_UART_SPEED),
         .uc_databits = 8,
         .uc_stopbits = 1,
         .uc_parity = UART_PARITY_NONE,
         .uc_flow_ctl = UART_FLOW_CTL_NONE,
-        .uc_tx_char = nmgr_uart_tx_char,
-        .uc_rx_char = nmgr_uart_rx_char,
-        .uc_cb_arg = nus
+        .uc_tx_char = smp_uart_tx_char,
+        .uc_rx_char = smp_uart_rx_char,
+        .uc_cb_arg = sus
     };
 
     /* Ensure this function only gets called by sysinit. */
     SYSINIT_ASSERT_ACTIVE();
 
-    rc = nmgr_transport_init(&nus->nus_transport, nmgr_uart_out, nmgr_uart_mtu);
+    rc = smp_transport_init(&sus->sus_transport, smp_uart_out, smp_uart_mtu);
```

Comment:
repos/apache-mynewt-core/mgmt/smp/transport/smp_uart/src/smp_uart.c:389:50: error: passing argument 2 of 'smp_transport_init' from incompatible pointer type [-Werror=incompatible-pointer-types]
     rc = smp_transport_init(&sus->sus_transport, smp_uart_out, smp_uart_mtu);
                                                  ^~~~~~~~~~~~
In file included from repos/apache-mynewt-core/mgmt/smp/transport/smp_uart/src/smp_uart.c:26:0:
repos/apache-mynewt-core/mgmt/smp/include/mynewt_smp/smp.h:66:5: note: expected 'smp_transport_out_func_t {aka int (*)(struct os_mbuf *)}' but argument is of type 'int (*)(struct smp_transport *, struct os_mbuf *)'
 int smp_transport_init(struct smp_transport *st,
     ^~~~~~~~~~~~~~~~~~

Reply:
removed smp_uart